src/vm/stubgen.cpp

```diff
@@ -269,30 +269,6 @@ ILCodeStream::ILInstrEnum ILCodeStream::LowerOpcode(ILInstrEnum instr, ILStubLin
 
     switch (instr)
     {
-        case CEE_LDC_I8:
-        {
-            if (uConst == (UINT_PTR)-1)
-            {
-                instr = CEE_LDC_I4_M1;
-            }
-            else
-            if (uConst < 9)
-            {
-                instr = (ILInstrEnum)((UINT_PTR)CEE_LDC_I4_0 + uConst);
-            }
-            else
-            if (FitsInI1(uConst))
-            {
-                instr = CEE_LDC_I4_S;
-            }
-            else
-            if (FitsInI4(uConst))
```

Comment:
I we wanted to keep the assert and most of the optimization, only removing this clause, would correspond to the original assert weakening.